ui/raidboss/timeline.js

```diff
@@ -199,45 +199,44 @@ export class Timeline {
         continue;
       const originalLine = line;
 
-      let match = regexes.ignore.exec(line);
-      if (match) {
-        this.ignores[match[1]] = true;
+      const [, ignoreId] = regexes.ignore.exec(line) || [];
+      if (ignoreId) {
+        this.ignores[ignoreId] = true;
         continue;
       }
 
-      match = regexes.tts.exec(line);
-      if (match) {
+      const [, ttsId, ttsSeconds, ttsCommand, ttsText] = regexes.tts.exec(line) || [];
+      if (ttsId && ttsSeconds && ttsCommand) {
         // TODO: Support alert sounds?
-        if (match[3] === 'sound')
+        if (ttsCommand === 'sound')
           continue;
-        texts[match[1]] = texts[match[1]] || [];
-        texts[match[1]].push({
+        texts[ttsId] = texts[ttsId] || [];
+        texts[ttsId].push({
           type: 'tts',
-          secondsBefore: parseFloat(match[2]),
-          text: match[4] ? match[4] : match[1],
+          secondsBefore: parseFloat(ttsSeconds),
+          text: ttsText ? ttsText : ttsId,
         });
         continue;
       }
-      match = regexes.soundAlert.exec(line);
+      let match = regexes.soundAlert.exec(line);
       if (match)
         continue;
       match = regexes.speaker.exec(line);
       if (match)
         continue;
 
-      match = regexes.popupText.exec(line);
-      if (match) {
-        texts[match[2]] = texts[match[2]] || [];
-        texts[match[2]].push({
-          type: match[1],
-          secondsBefore: parseFloat(match[3]),
-          text: match[4] ? match[4] : match[2],
+      const [, popupType, popupId, popupSeconds, popupText] = regexes.popupText.exec(line) || [];
+      if (popupType && popupId && popupSeconds) {
```

Comment:
(I had this brief thought of "what if popupSeconds is 0"? However, apparently even though `"0" == false` probably because it converts to a number first, `"0"` converted to a boolean is true.  [Thanks, JavaScript?!?!].)

That said, this style of conditional here and elsewhere doesn't allow for empty string, even though the regex allows for empty strings in a number of places.  I'm not saying it necessarily should just saying that this is a change in behavior in a number of places.  Do you think it'd be worth checking only defined here instead of truthy to convince TypeScript that they are valid?  Alternatively, I could go through and check them more carefully.